src/commands/multi_block/monitor.rs

```diff
@@ -201,10 +330,41 @@ where
         _ => return Ok(()),
     }
 
-    // 2. If the solution has already been submitted, nothing to do
-    if has_submitted(&storage, round, signer.account_id(), n_pages).await? {
+    // 2. If the solution has already been submitted:
```

Comment:
I would move this 2.1 and 2.2 to a separate function to make it a bit more readable and it's duplicated below, WDYT?